integration_test/sql/transaction.exs

```diff
@@ -205,6 +211,7 @@ defmodule Ecto.Integration.TransactionTest do
       end)
     end
 
+    @tag :transaction_checkout_raises
```

Comment:
So why this one does not fail? Is it related to the `handle_begin` change?

Reply:
This can't work since query should be "BEGIN TRAN"

one more thing, MSSQL supports nested transactions (nested in :transaction mode) :) but single rollback only works and @@TRANCOUNT must be check if you want to check if you are in transaction manually. On the other hand, savepoint can be rolled-back and it does not affect transaction count server variable and driver tracks it in its own state attribute.

BTW, I'm using Transaction Manager feature in TDS, instead of sending query, it is quicker since it is single call and does not require query to be compiled (it is not a query, just TDS message). Still, driver user can initiate "BEGIN TRAN" manually, but it is up to user to handle transaction status even tho TDS driver will register transaction ID when response to above.